src/Utils.cpp

```diff
@@ -905,3 +906,52 @@ void Utils::AddSourceHelper(void *_data, obs_scene_t *scene)
 	data->sceneItem = obs_scene_add(scene, data->source);
 	obs_sceneitem_set_visible(data->sceneItem, data->setVisible);
 }
+
+struct obs_data_item { // Used for OBSDataGetDefaults
+	volatile long ref;
+	struct obs_data *parent;
+	struct obs_data_item *next;
+	enum obs_data_type type;
+	size_t name_len;
+	size_t data_len;
+	size_t data_size;
+	size_t default_len;
+	size_t default_size;
+	size_t autoselect_size;
+	size_t capacity;
+};
+
+obs_data_t *Utils::OBSDataGetDefaults(obs_data_t *data)
+{
+	obs_data_t *returnData = obs_data_create();
+	obs_data_item_t *item = NULL;
+
+	for (item = obs_data_first(data); item; obs_data_item_next(&item)) {
+		enum obs_data_type type = obs_data_item_gettype(item);
+		const char *name = (char *)item + sizeof(struct obs_data_item);
```

Comment:
Isn't using `obs_data_item_get_name(obs_data_item_t*)` simpler than pulling off this hack?